Refactor AddressUtils to handle full country names

JorgeMucientes opened the pull request:
Fixes WOOMOB-1257

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
After recent change to remove deprecated usages of Locale() constructor a new crash was accidentally introduce in `AddressUtils` class. The following code: 

```kotlin
    fun getCountryLabelByCountryCode(countryCode: String): String {
        val locale = Locale.Builder()
            .setLanguage(Locale.getDefault().language)
            .setRegion(countryCode)
            .build()
        return locale.displayCountry
    }
```
Would lead to a crash when the passed value for `countryCode` is a country name al ready. 
```
java.util.IllformedLocaleException: Ill-formed region: India [at index 0]
	at java.util.Locale$Builder.setRegion(Locale.java:3316)
	at com.woocommerce.android.util.AddressUtils.getCountryLabelByCountryCode(AddressUtils.kt:15)
	at com.woocommerce.android.model.Order.formatShippingInformationForDisplay(Order.kt:310)
```

### Testing information
The unit tests added should be enough to ensure the function works as expected.

JorgeMucientes committed:
Refactor AddressUtils to handle full country names

The `getCountryLabelByCountryCode` method in `AddressUtils` has been updated to provide more robust handling of country inputs.

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/265">23.1 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes committed:
Fix unit tests style and fixes failing test

JorgeMucientes committed:
Fix detekt issues

Too many return statements
Test case not following correct style

hichamboushaba reviewed (approved):
LGTM, thanks @JorgeMucientes for the quick fix.

There is a warning in unit tests that prevents green CI.

JorgeMucientes committed:
Remove test name characters that could cause issues

This change fixes de following warning:
AddressUtilsTest.kt:47:9 Name contains character(s) that can cause problems on Windows: "
